FaunaDB.Client/Types/RefV.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using FaunaDB.Query;
+using Newtonsoft.Json;
+
+namespace FaunaDB.Types
+{
+    /// <summary>
+    /// A FaunaDB ref type.
+    /// <para>
+    /// See <see href="https://fauna.com/documentation/queries#values-special_types">FaunaDB Special Types</see>.
+    /// </para>
+    /// </summary>
+    public class RefV : ScalarValue<string>
+    {
+        /// <summary>
+        /// Create a Ref from a string, such as <c>new Ref("databases/prydain")</c>.
+        /// </summary>
+        /// <param name="value">Value.</param>
+        public RefV(string value) : base(value)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+        }
+
+        protected internal override void WriteJson(JsonWriter writer) =>
+            writer.WriteObject("@ref", Value);
+    }
+
+    public abstract class ObjectRef : RefV
```

Comment:
I'm sure core have reasons to treat them as different things. Specially because they might have different behaviour internally. But at the surface, references are simpler to reason about with less boiler plate, IMO.